quickwit/quickwit-search/src/collector.rs

```diff
@@ -638,6 +689,33 @@ impl Collector for QuickwitCollector {
             // this value isn't actually used.
             Ordering::Equal
         };
+        // Convert search_after into fast field u64
+        let mut search_after = self.search_after.clone();
```

Comment:
This piece of code has a few problems:

It use a functional style to mutate something.
It changes the semantics of something without adding a comment, or using a different type.

After this patch, search_after in Collector is a partial hit that contains sort values expressed in their original type.
and in the SegmentCollector, it uses their u64 representation.

Reply:
I replaced the functional style, and convert into a new `SearchAfterSegment` struct instead to make the conversion clear.